cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -53,17 +53,22 @@ folly::Future<folly::Unit> LocalVersionedEngine::delete_unreferenced_pruned_inde
 ) {
     try {
         if (!pruned_indexes.empty() && !cfg().write_options().delayed_deletes()) {
+            // TODO: the following function will load all snapshots, which will be horrifyingly inefficient when called
+            // multiple times from batch_*
             auto [not_in_snaps, in_snaps] = get_index_keys_partitioned_by_inclusion_in_snapshots(
                     store(),
                     pruned_indexes.begin()->id(),
                     pruned_indexes);
             in_snaps.insert(key_to_keep);
-            PreDeleteChecks checks{false, false, false, false, in_snaps};
-            return delete_trees_responsibly(not_in_snaps, {}, {}, checks);
+            PreDeleteChecks checks{false, false, false, false, std::move(in_snaps)};
+            return delete_trees_responsibly(not_in_snaps, {}, {}, checks)
+                    .thenError(folly::tag_t<std::exception>{}, [](auto const& ex) {
+                        log::version().warn("Failed to clean up pruned previous versions due to: {}", ex.what());
```

Comment:
Am I right in saying that for batch methods, this will result in an extra log, but a `VersionedItem` will still be returned rather than a `DataError`?

Reply:
Yes